### ISSUE TITLE
Transaction (BEGIN) passes linting without close (COMMIT)

### ISSUE THREAD

**Issue by KaffeMyers** [sbdchd/squawk#697]
Hi!

We just encountered a production issue during a migration where a SQL file contained a `BEGIN` statement without a corresponding `COMMIT`. Everything appeared to run successfully, but because the transaction was never committed, the changes didn’t apply, leading to a cascade of downstream issues.

Is there a reason why this isn’t flagged by Squawk? Should there be a check in place to catch unbalanced transaction blocks?

To prevent this from happening again, I’ve written a custom AWK-based linter that validates `BEGIN`/`COMMIT` pairing in migration files. Of course, having this built into Squawk would be ideal to avoid relying on custom tooling.💃

Thanks!

**Comment by sbdchd** [comment]
Seems like a good lint rule to have! 

Did you have an idea for a name? I'm thinking something like `ban-uncommitted-transactions`

**Comment by sbdchd** [comment]
Fixed in latest!


```
❯ echo 'begin; create table t(); ' | npx squawk-cli@latest
npm warn exec The following package was not found and will be installed: squawk-cli@2.30.0
warning[ban-uncommitted-transaction]: Transaction never committed or rolled back.
  ╭▸ stdin:1:1
  │
1 │ begin; create table t(); 
  │ ━━━━━
  │
  ├ help: Add a `COMMIT` or `ROLLBACK` statement to complete the transaction.
  ╭╴
1 ± 
2 + COMMIT;
  ╰╴

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 1 issue in 1 file (checked 1 source file)
```

**Comment by KaffeMyers** [comment]
That's some seriously fast turnaround!
Thanks a bunch. :)
